great_expectations/data_context/types/base.py

```diff
@@ -1049,7 +1055,13 @@ def validate_schema(self, data, **kwargs):
     # noinspection PyUnusedLocal
     @post_load
     def make_datasource_config(self, data, **kwargs):
-        return DatasourceConfig(**data)
+        # Add names to data connectors
+        data_copy: dict = copy.deepcopy(data)
```

Comment:
Do we truly need a deepcopy here? I don't think so and would try to avoid it due to the performance hit.

Reply:
I could go either way here. I think it adds some safety if `data` is used elsewhere. I don't have a strong preference.

Reply:
I'll remove it - if we need to ensure safety we can always add it back.

Reply:
hmm I'm pretty confident it isn't used after looking at the underlying Marshmallow code but I could go either way too!